common/isolationgroup/defaultisolationgroupstate/state.go

```diff
@@ -163,22 +169,45 @@ func (z *defaultIsolationGroupStateHandler) get(ctx context.Context, domain stri
 	return ig, nil
 }
 
+func (z *defaultIsolationGroupStateHandler) createAvailableisolationGroupMetricsScope(domainID string, availableIsolationGroups []string) metrics.Scope {
+	domainName, err := z.domainCache.GetDomainName(domainID)
+	if err != nil {
+		domainName = "unknown-domain"
+	}
+	return z.metricsClient.Scope(metrics.DefaultIsolationGroupStateHandlerGetAvailableIsolationGroupsScope).Tagged(metrics.DomainTag(domainName))
+}
+
 // A simple explicit deny-based isolation group implementation
-func availableIG(allIsolationGroups []string, global types.IsolationGroupConfiguration, domain types.IsolationGroupConfiguration) types.IsolationGroupConfiguration {
+func availableIG(
+	allIsolationGroups []string,
+	availablePollers types.IsolationGroupConfiguration,
+	global types.IsolationGroupConfiguration,
+	domain types.IsolationGroupConfiguration,
+	scope metrics.Scope,
+) types.IsolationGroupConfiguration {
 	out := types.IsolationGroupConfiguration{}
 	for _, isolationGroup := range allIsolationGroups {
+		_, hasAvailablePollers := availablePollers[isolationGroup]
 		globalCfg, hasGlobalConfig := global[isolationGroup]
 		domainCfg, hasDomainConfig := domain[isolationGroup]
+		if !hasAvailablePollers {
```

Comment:
check drain before poller. Because we block poller requests in frontend if it's drained

Reply:
This function isn't used in the `IsDrained`, it doesn't make sense to look at where pollers are in blocking requests in frontend because there's no pollers there.

Reply:
If an isolation group is drained, you won't see `IsolationGroupStateDrained` metric. You'll just see `IsolationGroupStatePollerUnavailable` metric.

Reply:
oh, I misunderstood your point. Yeah, that's a fair point